ambry-api/src/main/java/com/github/ambry/named/NamedBlobRecord.java

```diff
@@ -110,7 +111,9 @@ public boolean equals(Object o) {
       return false;
     }
     NamedBlobRecord record = (NamedBlobRecord) o;
-    return expirationTimeMs == record.expirationTimeMs && Objects.equals(accountName, record.accountName)
```

Comment:
Without this change, the 'MySqlNamedBlobDbIntegrationTest/testPutGetListDeleteSequence' will fail at below line: 
`      assertEquals("Record does not match expectations.", record, recordFromStore);`

due to the fact that original **record** has expirationTimeMs=-1, but the delete operation, the **recordFromStore** will have the real deletes_ts value as expirationTimeMs. 
Plus for a 'NamedBlobRecord' as long as the other key attributes matches, we can treat it as equal, similar to MySQL primary keys matches, then we don't check other non-primary key fields.

```
<img width="1683" alt="Screen Shot 2022-08-18 at 6 01 55 PM" src="https://user-images.githubusercontent.com/102339297/185503010-19d6ad56-bd25-4b4d-a936-8cc74ce1b094.png">
```

Reply:
Based on my understanding, a ttled blob and a permanent blob can not consider as equal because you are comparing the get blob between old vs new DB. It might cause some data loss especially something wrong during the migration. I suggest we instead of compare record with recordFromStore, we can compare each field(excluding expirationTimeMs and version).

Reply:
So you are saying in the unit test we do something like below? But then that's just a more verbose way of doing the record == recordFromStore? I don't see the NamedBlobRecord equality check done elsewhere except in the unit tests, so I think removing the 'expirationTimeMs' from equality check should be fine. 
```
assertEquals("...", record.getAccountName(), recordFromStore.getAccountName());
assertEquals("...", record.getContainerName(), recordFromStore.getContainerName());
assertEquals("...", record.getBlobName(), recordFromStore.getBlobName());
......
```